scalafix-interfaces/src/main/java/scalafix/interfaces/ScalafixMainCallback.java

```diff
@@ -0,0 +1,13 @@
+package scalafix.interfaces;
+
+/**
+ * Callback handler for events that happen in the command-line interface.
+ */
+public interface ScalafixMainCallback {
+    /**
+     * Handle a diagnostic event reported by the Scalafix reporter.
+     *
+     * @param diagnostic the reported diagnostic.
+     */
+    void reportDiagnostic(ScalafixDiagnostic diagnostic);
```

Comment:
This is a bit limited compared to what scalafix can do.

It would be useful to have a list of files that were modified. Ideally, it would be useful to have interactive rewrites where the user can make a choice. For example for collection.Seq in the new collection design.